MainPage View (SeongGeun)

ParkSeongGeun opened the pull request:
### 구현할 화면
<img width="241" alt="image" src="https://github.com/gdsc-konkuk/iOS-DragDrop/assets/117553364/167c2af6-3788-4fe1-9b77-676d11fb951d">

#### 참고할점
- 주석으로 첨부했습니다.
- 추가로, TaskCard View에 대한 논의를 한 뒤에 마저 수정해야 할 거 같습니다. (우선은, MainPage.swift 내에 따로 View를 생성했습니다)
- 버튼을 눌렀을 때, 어떻게 메뉴들이 출력되는 지 정해지지 않았다고 판단해 Menu를 이용해서 간단하게 구현했습니다.
- <U>글씨 크기 수정해야합니다.</U>
- NavigationView로 한 거는 저 화면을 통해서 타고 들어가는 화면이 있다고 판단하여 사용했습니다.
- 이때, NavigationView는 아래로 스와이프 했을 때, iOS 15 이상에서 도입된 "scroll edge appearance" 때문이라고 합니다.

### 실행화면


https://github.com/gdsc-konkuk/iOS-DragDrop/assets/117553364/37eda908-b032-44fb-9f50-6c922b5d13da

ParkSeongGeun committed:
Update MainPage.swift

- TaskCard View 수정했다가 원본으로 두었습니다.
- tasks는 테스트용 배열입니다.
- List -> ForEach구문 위에 NavigationView 추가해줬습니다.
- 화면 우측 상단 버튼을 Menu를 이용해서 구현했는데 추후 의논 후 반영하겠습니다.

ParkSeongGeun committed:
MainPageTaskCard View 생성

ParkSeongGeun committed:
FontSize Fix

sujileelea commented on DragDrop/DragDrop/View/MainPage.swift:
```diff
@@ -8,26 +8,77 @@
+struct MainPageTaskCard: View {
+    var task: Task
+    var body: some View {
+        HStack {
```
아이콘과 글자의 간격은 spacing으로 조절하고, HStack 자체에 padding 수정자를 써서 요소의 앞 뒤 간격을 조정하는 건 어떠세요?

sujileelea commented on DragDrop/DragDrop/View/MainPage.swift:
```diff
@@ -8,26 +8,77 @@
+            .padding()
+            
+            VStack {
+                Menu {
```
이 부분은 후에 병직님이 만든 커스텀 뷰로 대체하겠습니다.

sujileelea commented on DragDrop/DragDrop/View/MainPage.swift:
```diff
@@ -8,26 +8,77 @@
+            .offset(x: screenWidth * 0.4, y: screenHeight * -0.03)
+        }
+        
+        NavigationView {
```
ToolbarItem을 위한 NavigationView로 보이는데 edit은 그저 버튼이고, ToolbarItem은 화면 우측 상단의 사이드스크린 토글 버튼입니다.

ParkSeongGeun committed:
Update MainPage.swift

PR Review applied
- Toolbar item을 화면에 보이게 하기 위해 최상단에 NavigationView를 height 50만큼 생성했습니다.
- 추가 comment 반영했습니다.

ParkSeongGeun commented:
PR review 반영했습니다😌

sujileelea commented:
깃 이슈 문제로 코드 보존 후 close 합니다.
